src/coreComponents/mesh/MeshLevel.cpp

```diff
@@ -121,35 +119,43 @@ MeshLevel::MeshLevel( string const & name,
   m_faceManager->faceNormal() = source.m_faceManager->faceNormal();
 
   // Faces
-//  ArrayOfArraysView< localIndex const > const facesToNodesMapSource = m_faceManager->nodeList().toViewConst();
   ArrayOfArrays< localIndex > & faceToNodeMapNew = m_faceManager->nodeList();
   ArrayOfArraysView< localIndex const > const & facesToEdges = m_faceManager->edgeList().toViewConst();
-  localIndex const estimatedNumNodesPerFace = pow( order+1, 2 );
-  faceToNodeMapNew.resize( faceToNodeMapNew.size(), estimatedNumNodesPerFace );
+  localIndex const numNodesPerFace = pow( order+1, 2 );
 
 
   // add the number of non-edge face nodes
   localIndex numInternalFaceNodes = 0;
+  localIndex const numNonEdgeNodesPerFace = pow( order-1, 2 );
   for( localIndex kf=0; kf<m_faceManager->size(); ++kf )
   {
     localIndex const numEdgesPerFace = facesToEdges.sizeOfArray( kf );
-//    localIndex const numVertexNodesPerFace = facesToNodesMapSource.sizeOfArray( kf );
-//    localIndex const numEdgeNodesPerFace = numVertexNodesPerFace + numEdgesPerFace * numNonVertexNodesPerEdge;
-
     if( numEdgesPerFace==4 )
-    {
-      localIndex const numNonEdgeNodesPerFace = pow( order-1, 2 );
       numInternalFaceNodes += numNonEdgeNodesPerFace;
-
-      localIndex const numNodesPerFace = pow( order+1, 2 );
-      faceToNodeMapNew.resizeArray( kf, numNodesPerFace );
-    }
     else
     {
       GEOSX_ERROR( "need more support for face geometry" );
     }
   }
 
+  // number of elements in each row of the map as capacity
+  array1d< localIndex > counts( faceToNodeMapNew.size());
+  counts.setValues< parallelHostPolicy >( numNodesPerFace );
```

Comment:
The intent of original code was that cases other than `numEdgesPerFace==4` would eventually be handled for meshes with other element/face shapes. This refactoring hardcodes face shape to be constant by moving certain calculations out of the `numEdgesPerFace==4` branch. I think we should revert some of these changes while keeping the main optimization (one-time allocation of the map) in place. In particular, the values of `counts` should be set inside the loop above based on each individual face shape.

Reply:
I see, thanks a lot for clarifying the background story. Yes, it would be great to make changes to adapt the code to general cases.